Ported/combat-bees/Assets/Scripts/Systems/FoodSpawnerSystem.cs

```diff
@@ -46,17 +51,30 @@ public void OnUpdate(ref SystemState state)
             ecb.SetComponent(food, tm);
         }
 #else
-        var foodSpawnJob = new SpawnJob
+        var nestEntities = NestQuery.ToEntityArray(Allocator.Temp);
+
+        foreach (var nest in nestEntities)
         {
-            // Note the function call required to get a parallel writer for an EntityCommandBuffer.
-            ECB = ecb.AsParallelWriter(),
-            Prefab = config.food,
-            InitTM = state.EntityManager.GetComponentData<LocalToWorldTransform>(config.food)
-        };
+            var nestFaction = state.EntityManager.GetComponentData<Faction>(nest);
+            if (nestFaction.Value != (int)Factions.None)
```

Comment:
Factions None feels unnecessary.  It's possible to check for existence of the component using state.EntityManager.HasComponent.  If you don't want to do that it makes more sense to split the query into one with faction (spawns bees) and one without (spawns food)